src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,650 @@
+            checkIfRequestForItemExistOrCreateAsynch(item);
+        }
 
-            for (int i = 0; i < workBuilding.getTileEntity().getSizeInventory(); i++)
+        if (getOwnBuilding() != null)
+        {
+            final TileEntityColonyBuilding chest = getOwnBuilding().getTileEntity();
+            for (int i = 0; i < getOwnBuilding().getTileEntity().getSizeInventory(); i++)
             {
                 final ItemStack stack = chest.getStackInSlot(i);
 
-                if (ItemStackUtils.isEmpty(stack))
-                {
-                    continue;
-                }
-
-                if (stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType) == null)
+                if (InventoryUtils.findFirstSlotInProviderWith(chest,
+                  itemStack -> itemStack.getItem() instanceof ItemArmor) != -1)
                 {
-                    final int emptySlot = InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(worker.getInventoryCitizen()),
-                      ItemStackUtils::isEmpty);
-
-                    if (emptySlot != -1)
-                    {
-                        new InvWrapper(worker.getInventoryCitizen()).insertItem(emptySlot, stack, false);
-                        chest.setInventorySlotContents(i, ItemStackUtils.EMPTY);
-                    }
+                    InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(
+                      getOwnBuilding(), itemStack -> itemStack.getItem() instanceof ItemArmor,
+                      stack.getCount(),
+                      new InvWrapper(worker.getInventoryCitizen()));
                 }
-                dumpAfterActions = DUMP_BASE * workBuilding.getBuildingLevel();
             }
         }
-        attacksExecuted = 0;
-        return GUARD_SEARCH_TARGET;
-    }
 
-    @Override
-    protected int getActionsDoneUntilDumping()
-    {
-        return dumpAfterActions;
+        return DECIDE;
     }
 
     /**
-     * Can be overridden in implementations.
-     * <p>
-     * Here the AI can check if the armour have to be re rendered and do it.
-     */
-    @Override
-    protected void updateRenderMetaData()
-    {
-        updateArmor();
-    }
-
-    /**
-     * Updates the equipment. Always take the first item of each type and set it.
+     * Decide what we should do next!
+     *
+     * @return the next AIState.
      */
-    protected void updateArmor()
+    protected AIState decide()
     {
-        worker.setItemStackToSlot(EntityEquipmentSlot.CHEST, ItemStackUtils.EMPTY);
-        worker.setItemStackToSlot(EntityEquipmentSlot.FEET, ItemStackUtils.EMPTY);
-        worker.setItemStackToSlot(EntityEquipmentSlot.HEAD, ItemStackUtils.EMPTY);
-        worker.setItemStackToSlot(EntityEquipmentSlot.LEGS, ItemStackUtils.EMPTY);
-
-        for (int i = 0; i < new InvWrapper(worker.getInventoryCitizen()).getSlots(); i++)
+        setDelay(20);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)